src/errors/resolutionError.ts

```diff
@@ -26,6 +27,7 @@ export enum ResolutionErrorCode {
   IncorrectResolverInterface = 'IncorrectResolverInterface',
   RecordNotFound = 'RecordNotFound',
   ServiceProviderError = 'ServiceProviderError',
+  NetworkError = 'NetworkError',
```

Comment:
@sudoryan We can't know for sure that error is caused by network issues. The only think we can do - is to detect known errors and fallback to UnkownError with low-level error details if we couldn't determine issue root cause if the issue.